src/components/network/geo-data.js

```diff
@@ -88,4 +90,67 @@ export default class GeoData {
 
         return [ substationPosition1, substationPosition2 ];
     };
+
+    getLineDistances(positions) {
+        if (positions !== null && positions.length > 1) {
+            let cumulativeDistanceArray = [0];
+            let cumulativeDistance = 0;
+            let segmentDistance;
+            let ruler;
+            for (let i = 0; i < positions.length - 1; i++) {
+                ruler = cheapRuler(positions[i][1], 'meters');
+                segmentDistance = ruler.lineDistance(positions.slice(i, i + 2));
+                cumulativeDistance = cumulativeDistance + segmentDistance;
+                cumulativeDistanceArray[i+1] = cumulativeDistance;
+            }
+            return cumulativeDistanceArray;
+        }
+        return null;
+    }
+
+    getCoordinateInLine(positions, cumulativeDistances, percent) {
```

Comment:
rename to "labelDisplayPosition" or something like this (it only works for the label because it has the offset)